### ISSUE TITLE
Cannot change primary key field

### ISSUE THREAD

**Issue by StevenHit** [hougaard/ToolBox-Issues#18]
Hi,

We try to change the document number for a posted sales invoice via ToolBox, but we get errors.

Is there a possibility to rename records with any of the toolbox functionalities?

Thanks

![image](https://github.com/hougaard/ToolBox-Issues/assets/34163966/8b309102-4691-4356-9ff9-535f996b8bdf)

![image](https://github.com/hougaard/ToolBox-Issues/assets/34163966/20bb8d7c-f64b-4567-8fb4-1eea57ffa1c3)

**Comment by hougaard** [comment]
No, not possible in the batch processing module, you'll have to write AL code to do that.

**Comment by StevenHit** [comment]
The coding feature in Toolbox doesn't seem to have a "rename" command. Does that mean that we cannot rename with any of the toolbox features?

**Comment by hougaard** [comment]
Yeah, .Rename() is complicated to implement behind the scene in a truly dynamic manner. 

I usually copy the record with rec.copy() and the .delete() and .Insert().

I might be able to implement .Rename() for single field key with Code fields..
